### ISSUE TITLE
SVG Style adjustment to position:relative

### ISSUE THREAD

**Issue by Ocmeramutna** [huww98/TimeChart#14]
How can I adjust the SVG CSS Shadow Root to `position: relative`? 
Now, the chart takes only half of the parent div.. 
I am using Vue. 
Thank you a lot in advance.

The picture below shows my problem:
![image](https://user-images.githubusercontent.com/64258625/100545075-4a4df300-325a-11eb-909f-19335b7de0c2.png)
The picture below shows show how to fix it:
![image](https://user-images.githubusercontent.com/64258625/100545023-f6dba500-3259-11eb-9314-f746eaef0559.png)
The picture below shows the result of the CSS adjustment:
![image](https://user-images.githubusercontent.com/64258625/100545099-749fb080-325a-11eb-84dd-eb843052f00a.png)

**Comment by huww98** [comment]
Can you provide a demo so that I can reproduce this? Under what circumstance did you see this behavior?

Maybe calling `chart.onResize()` will be sufficient to fix this. Setup a button and call `onResize` in the listener, and see if it is the case.

**Comment by Ocmeramutna** [comment]
Below the code: 
```
<template>
    <div class="ParentDiv">
        <div ref="TimeChartRef" class="TimeChartClass"></div>
    </div>
</template>

<script>
import TimeChart from "timechart"
import { scaleLinear } from "d3-scale"

export default {
    name: "NotImportant",
    data: function() {
        return {
            UpdateInterval: 15,

            Data1: 0,
            Data2: 0,
            Data3: 0,

            chart: null,

            ShowGraph: true,
        }
    },
    mounted() {
        //Initialize chart
        if(this.chart == null) {
            this.chart = new TimeChart(this.$refs.TimeChartRef, {
                series: 
                [
                    {
                        name: "Data1",
                        lineWidth: 5,
                        color: 'white',
                    },
                    {
                        name: "Data2",
                        lineWidth: 5,
                        color: 'white',
                    },
                    {
                        name: "Data3",
                        lineWidth: 8,
                        color: 'purple',
                    }
                ],
                realTime: true,
                xScaleType: scaleLinear,
                xRange: {
                    min: 0, 
                    max: 10000
                },
            })
        }

        //Show Chart
            setTimeout(() => {
                this.chart.onResize()
            }, 1000);

        //Set update interval
        this.DataInterval = setInterval(() => this.UpdateChartFunction(), this.UpdateInterval); 
    },
    methods: {
        UpdateChartFunction() {
            this.Data3 += 2;
            this.Data1 += 5;
            this.Data2 += 15;
        
            this.chart.options.series[0].data.push(                    
                {
                    x: this.chart.options.series[0].data.length * this.UpdateInterval, 
                    y: this.Data1
                }
            );
            this.chart.options.series[1].data.push(
                {
                    x: this.chart.options.series[1].data.length * this.UpdateInterval,
                    y: this.Data2
                }
            );
            this.chart.options.series[2].data.push(
                {
                    x: this.chart.options.series[2].data.length * this.UpdateInterval,
                    y: this.Data3
                }
            );
            this.chart.update()
        },
    },
};
</script>
<style scoped>
.ParentDiv {
    box-sizing: border-box;
    position: absolute;
    top: 120px;
    left: 15px;
    display: block;
    width: 1350px;
    height: 480px;
    /* border: 2px solid red; */
}
.TimeChartClass {
    height: 100%;
    width: 100%;
    color: white;
    background-color: black;
    --background-overlay: black;
}
</style>
```
this.chart.onResize() fixes another problem that I had (chart didn't show up until I zoomed in / out on the window), but unfortunately not the 'half ParentDiv fill' problem.. 
As I said, changing in Chrome DevTools the position to 'relative' of the SVG in the CSS Shadow-Root solves my problem, but I don't know how I can do this in the code.

I implemented the graph in the meantime and it is super fast. It took a lot of time to find a fast library, thanks a lot for that. 
Besides the problem described above, I have another question / problem:
Using `realtime: true` causes a data visualization delay of approximately 9000 ms. See the screen video below for an explanation.
[ScreenVideo.zip](https://github.com/huww98/TimeChart/files/5612554/ScreenVideo.zip)
Can I adjust this delay value?

**Comment by Ocmeramutna** [comment]
The code below fixed my problem first problem (filling half of parent div):
`               
this.$refs.ParentDivRef.querySelectorAll("div.TimeChartClass")[0].shadowRoot.querySelector("svg").style.cssText = "position: relative; width: 100%; height: 100%;"
`

**Comment by huww98** [comment]
Sorry for the delay. I've just published v0.5.2 which should fix this issue. The delay you saw is likely also caused by the issue of the misplaced elements.
